aws-lambda-java-runtime-interface-client/src/main/jni/build-jni-lib.sh

```diff
@@ -7,10 +7,14 @@ SRC_DIR=$(dirname "$0")
 DST_DIR=${1}
 CURL_VERSION=7.77.0
 
-# compile the native library
-docker build -f "${SRC_DIR}/Dockerfile.glibc" --build-arg CURL_VERSION=${CURL_VERSION} -t lambda-java-jni-lib-glibc "${SRC_DIR}"
-docker run --rm --entrypoint /bin/cat lambda-java-jni-lib-glibc /src/aws-lambda-runtime-interface-client.so > "${DST_DIR}"/classes/aws-lambda-runtime-interface-client.glibc.so
+for i in x86,linux/amd64 arm,linux/arm64/v8; do
+  arch=${i%,*};
+  platform=${i#*,};
 
-docker build -f "${SRC_DIR}/Dockerfile.musl" --build-arg CURL_VERSION=${CURL_VERSION} -t lambda-java-jni-lib-musl "${SRC_DIR}"
-docker run --rm --entrypoint /bin/cat lambda-java-jni-lib-musl /src/aws-lambda-runtime-interface-client.so > "${DST_DIR}"/classes/aws-lambda-runtime-interface-client.musl.so
+  mkdir "${DST_DIR}"/classes/$arch
+  docker build --platform=$platform -f "${SRC_DIR}/Dockerfile.glibc" --build-arg CURL_VERSION=${CURL_VERSION} -t lambda-java-jni-lib-glibc-$arch "${SRC_DIR}"
+  docker run --rm --entrypoint /bin/cat lambda-java-jni-lib-glibc-$arch /src/aws-lambda-runtime-interface-client.so > "${DST_DIR}"/classes/$arch/aws-lambda-runtime-interface-client.glibc.so
 
+  docker build --platform=$platform -f "${SRC_DIR}/Dockerfile.musl" --build-arg CURL_VERSION=${CURL_VERSION} -t lambda-java-jni-lib-musl-$arch "${SRC_DIR}"
+  docker run --rm --entrypoint /bin/cat lambda-java-jni-lib-musl-$arch /src/aws-lambda-runtime-interface-client.so > "${DST_DIR}"/classes/$arch/aws-lambda-runtime-interface-client.musl.so
+done
```

Comment:
Variables suggested in https://github.com/carlzogh/aws-lambda-java-libs/commit/0b16ecde364fe03ea5036a9301581a1cfb9b9797 help keep our scripts more maintainable, inline variable mappings are not very easy to follow.

Reply:
incorporated in the last update